File: algorithms/gr-prims/C++/is_prime.cpp
#include <iostream>
using namespace std;


bool is_prime(int n) {
    int d = 2;
    bool primo;
    if (n <= 1) return false;
    else {
        primo = true;
        while (primo && d <= n/2) {
            if (n % d == 0) {
                primo =false;
            }
            d++;
        } return primo;
    }
}


int main()
{
    int n;
    cout<<"Enter the number: ";
    cin>>n;
    if (is_prime(n)) {
        cout<<"Prime"<<endl;
    } else cout<<"\nIsn't prime"<<endl;
    return 0;
}

File: algorithms/gr-prims/javascript/isPrime.js
// program to check if a number is prime or not

function isPrime(n) {
	let flag = true

	for (let i = 2; i <= n - 1; i++)
		if (n % i == 0) {
			flag = false
			break
		}

	if (flag) return true

	return false
}

console.log(isPrime(2)) // true
console.log(isPrime(4)) // false
console.log(isPrime(7)) // true
